### DISCUSSION TITLE
Is it possible to add scalars to a mesh without making them active?

### DISCUSSION THREAD

**Discussion by p-j-smith** [pyvista/pyvista#3068] [Q&A] [answered]
Hi,

I'm wondering whether it's possible to add scalars to a mesh without making them active?

e.g.:

```py
import pyvista as pv

sphere = pv.Sphere()
sphere.point_data['zeros'] = np.zeros(sphere.n_points)
sphere.point_data['ones'] = np.ones(sphere.n_points)
```

In the above example, 'ones' are the active scalars. However, I'd like 'zeros' to remain the active scalars even after adding ones. Is it possible? Or should I just re-set the active scalars after adding all point data?

**Comment by akaszynski** [comment, marked as answer]
Good question.

You can do it with [set_array](https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetAttributes.set_array.html)

Continuing your example:

```py
>>> import pyvista as pv
>>> import numpy as np
>>> sphere = pv.Sphere()
>>> sphere.point_data.set_array(np.zeros(sphere.n_points), 'zeros')
>>> sphere.point_data.set_array(np.ones(sphere.n_points), 'ones')
>>> sphere.point_data
pyvista DataSetAttributes
Association     : POINT
Active Scalars  : None
Active Vectors  : None
Active Texture  : None
Active Normals  : Normals
Contains arrays :
    Normals                 float32    (842, 3)             NORMALS
    zeros                   float64    (842,)
    ones                    float64    (842,)
```

**Reply by p-j-smith** [reply]
Ah that's brilliant, thank you!

**Reply by akaszynski** [reply]
Thanks for posting the question here. It will help others who have the same question as you.
